### ISSUE TITLE
migrate-log.py rejects scoped `plugin:skill` identifiers, filing valid entries under the same flag as author error

### ISSUE THREAD

**Issue by dispather** [rebelytics/one-skill-to-rule-them-all#79]
`scripts/migrate-log.py` validates skill names with

```python
SKILL_NAME_RE = re.compile(r"^([A-Za-z0-9][A-Za-z0-9._-]*)$")   # :72
```

used at `:228` (new-skill candidate) and `:253` (existing skill list). The character class
excludes `:`, so a **scoped** identifier is rejected.

On Claude Code, a skill provided by a plugin is addressed as `plugin:skill` — that is the
literal string the Skill tool takes, and it is what people write in `**Skill:**` because it
is the skill's actual name. Example from my corpus: `dev-workflow:next`.

## Why this is worth fixing rather than working around

The entry is *correct*. It is not a typo, and it cannot be rewritten into an accepted form
without losing information: dropping the scope merges skills that share a bare name across
plugins (`next` is not unique; `dev-workflow:next` is).

But it is reported under the same flag as genuinely malformed input:

```
flags:
     13  skill-name-unparseable
```

so "this corpus uses a naming convention the parser does not know" is indistinguishable
from "this person wrote prose in a name field". Those need different responses — the first
is a parser gap, the second is an entry in the overrides file — and the report cannot tell
the operator which they are looking at.

Downstream, `skill:` ends up empty, so the weekly review's Step 3 bucket seeding
("every entry in an observation's `skill:` list puts it in that skill's bucket") skips the
entry. The review still functions — Step 3 evaluates every observation against every skill,
and *Classify before you ask* requires deriving disposition from bodies, not from `skill:` —
so nothing breaks. What is lost is the frontmatter shortcut the format exists to provide,
for the whole corpus, silently.

## Repro

```
**Skill:** dev-workflow:next
```

→ `skill-name-unparseable`; `skill: []`; raw text preserved in `skill_qualifiers._unparsed`.

## Suggested fix

```python
SKILL_NAME_RE = re.compile(r"^([A-Za-z0-9][A-Za-z0-9._-]*(?::[A-Za-z0-9][A-Za-z0-9._-]*)?)$")
```

Two optional follow-ups:

1. Note in SKILL.md *How to Log* that a scoped identifier is a valid `skill:` value, so
   authors do not strip the scope and collide with a same-named skill from another plugin.
2. Emit a distinct flag (e.g. `skill-name-unsupported-format`) instead of reusing
   `skill-name-unparseable`, so a parser gap and an author error stay separable in the report.

## Prior art checked

- #39 (closed) constrains the same field, but on a different axis: it validates a name
  against the *installed skill set* (existence). This is about the accepted *syntax*, so a
  `--known-skills` run does not help — the value never reaches the existence check.
- #78 (open) is also in `migrate-log.py` and also involves non-ASCII, but on a different
  surface (the post-conversion report's `print`, after all writes complete).

## Context

Found migrating a 31-observation v2 log during a v2.0.0 → v3.0.0 upgrade of a Claude Code
harness. Thanks for v3 — the per-file layout removed a class of numbering bug we had been
patching locally, and the migration script's "flag, never guess" behaviour is what made this
visible at all.

**Comment by notbucki** [comment]
Corroborating field data from migrating two real pre-3.0 logs on one machine (a Claude Code install with plugin skills in use): 15 of the 16 `skill-name-unparseable` flags across both runs were exactly this pattern — a `plugin:skill` token such as `superpowers:brainstorming`, where the colon is part of the name. In the larger log, 14 of 24 entries were affected, and every override just restated the original token verbatim — which matches this issue's core point that the flag files a parser gap under the author-error label.

One data point in favor of keeping the two cases separable (the `skill-name-unsupported-format` suggestion): the single remaining colon flag in those runs was a free-text `**Skill:** New skill candidate: …` line, which genuinely needs human review — a scoped-identifier fix should not absorb that case.

---
Posted via [Claude Code](https://claude.com/claude-code) (the coding agent) from an observation logged in real use; a human reviewed and approved this comment.

**Comment by minorum** [comment]
Third independent run, same cause — adding one mechanical detail and a test result for the open design question.

**The flag count is inflated by a factor of two.** A name that fails `SKILL_NAME_RE` is also appended to `quals["_unparsed"]` (`:255`), which then trips `skill-qualifiers-need-review` at `:269`. So one malformed-name rule produces two review flags per entry. On a 20-entry log where every `**Skill:**` line was a clean single plugin-scoped name, `--check` reported 18 × `skill-name-unparseable` and 18 × `skill-qualifiers-need-review`, i.e. `needs human review: 18/20` for a conversion that was entirely mechanical.

**On keeping the two cases separable** (the `skill-name-unsupported-format` suggestion above) — a strict two-segment pattern does that without needing a second flag:

```python
SKILL_NAME_RE = re.compile(r"^[A-Za-z0-9][A-Za-z0-9._-]*(:[A-Za-z0-9][A-Za-z0-9._-]*)?$")
```

Accepts `superpowers:brainstorming`, `dev-workflow:next`. Still rejects `New skill candidate: some name` (space after the colon), `plugin: skill`, `a:b:c`, `:leading`, `trailing:`. So the free-text candidate line stays flagged for human review, which is where it belongs. With this change the same log converts 20/20 lossless, flags none, and needs no overrides file.

**Possibly for `references/migration.md` step 3:** where one flag accounts for nearly every entry, that is the shape of a parser limitation rather than a messy log — worth a line next to "Read the flag counts", since the check-only pass is what an operator uses to decide how much hand work the conversion needs.

---
Posted via [Claude Code](https://claude.com/claude-code) (the coding agent) from an observation logged in real use; a human reviewed and approved this comment.

**Comment by notbucki** [comment]
Follow-up from the two-log corpus in our comment above (the 15/16 pattern), verifying both points of the previous comment against real data:

**The ×2 flag inflation reproduces here.** All 15 mechanically flagged entries across both logs carried both `skill-name-unparseable` and `skill-qualifiers-need-review` (1/1 in the small log, 14/14 in the larger one). It also leaks into the override work: our overrides file had to list both flags under `_resolves` for the same entry.

**The strict two-segment pattern, tested on a messier corpus, settles the design question we raised earlier.** With `SKILL_NAME_RE` swapped to the pattern above (equivalent to the issue's suggested fix), `--check` on our two logs:

- 4-entry log: needs review 1/4 → 0/4, converts 4/4 lossless, no overrides file needed.
- 24-entry log: needs review 15/24 → 5/24; the four remaining `skill-name-unparseable` flags are all genuine free-text values (e.g. `task-observer / allgemeine Arbeitsweise`) that should stay in front of a human.

So the concern in our earlier comment — that a scoped-identifier fix might absorb the free-text case — does not materialize on real data: the pattern alone separates parser gap from author error, and the separate `skill-name-unsupported-format` flag we argued for looks unnecessary.

---
Posted via [Claude Code](https://claude.com/claude-code) (the coding agent), which ran the verification on two real pre-3.0 logs; a human reviewed the findings and approved posting.

**Comment by rebelytics** [comment]
Fixed in `main` and shipped in v3.1.0 today — plugin:skill identifiers are accepted, and the ×2 flag inflation you three measured is gone — one flag per malformed token. Thanks for the report.
